custom_components/homewhiz/select.py

```diff
@@ -83,81 +92,49 @@ def __init__(self, program: ApplianceProgram):
         )
 
 
-class SubProgramBoundedEntityDescription(HomeWhizEntitySelectDescription):
-    def __init__(
-        self,
-        parent_key: str,
-        bounds: ApplianceFeatureBoundedOption,
-        read_index: int,
-        write_index: int,
-    ):
-        self.key = bounds.strKey if bounds.strKey else parent_key
-        self._bounds = bounds
-        self.read_index = read_index
-        self.write_index = write_index
-        self.options = [
-            str(num)
-            for num in range(bounds.lowerLimit, bounds.upperLimit + 1, bounds.step)
-        ]
-
-    def byte_to_option(self, value):
-        return value * self._bounds.factor
-
-    def option_to_byte(self, value):
-        return int(value)
-
-    @property
-    def native_unit_of_measurement(self):
-        if "TEMP" in self.key:
-            return TEMP_CELSIUS
-        if "SPIN" in self.key:
-            return "rpm"
-
-    @property
-    def icon(self):
-        if "TEMP" in self.key:
-            return "mdi:thermometer"
-        if "SPIN" in self.key:
-            return "mdi:rotate-3d-variant"
-
-
-def generate_descriptions_from_features(features: list[ApplianceFeature]):
+def generate_select_descriptions_from_features(features: list[ApplianceFeature]):
     result = []
     for feature in features:
-        read_index = feature.wifiArrayIndex
-        write_index = (
-            feature.wfaWriteIndex
-            if feature.wfaWriteIndex is not None
-            else feature.wifiArrayIndex
-        )
-        if feature.boundedValues is not None:
-            for bounds in feature.boundedValues:
-                result.append(
-                    SubProgramBoundedEntityDescription(
-                        feature.strKey, bounds, read_index, write_index
-                    )
-                )
+        options: dict[int, str] = {}
         if feature.enumValues is not None:
-            result.append(
-                EnumEntityDescription(
-                    feature.strKey,
-                    feature.enumValues,
-                    read_index,
-                    write_index,
-                )
+            options = options | {
+                option.wifiArrayValue: option.strKey for option in feature.enumValues
+            }
+        if feature.boundedValues is not None:
+            for boundedValues in feature.boundedValues:
+                value = boundedValues.lowerLimit
+                while value <= boundedValues.upperLimit:
```

Comment:
```python3
for value in range(boundedValues.lowerLimit, boundedValues.upperLimit, boundedValues.step):
```

Reply:
step is a float, range operates only on ints. 

we could use numpy, but I feel it would be an overkill